reflex/utils/format.py

```diff
@@ -311,6 +330,46 @@ def format_event(event_spec: EventSpec) -> str:
     return f"E({', '.join(event_args)})"
 
 
+def format_event_chain(
+    event_chain: EventChain | Var[EventChain],
+    event_arg: Var | None = None,
+) -> str:
+    """Format an event chain as a javascript invocation.
+
+    Args:
+        event_chain: The event chain to queue on the frontend.
+        event_arg: The browser-native event (only used to preventDefault).
+
+    Returns:
+        Compiled javascript code to queue the given event chain on the frontend.
+
+    Raises:
+        ValueError: When the given event chain is not a valid event chain.
+    """
+    if isinstance(event_chain, Var):
+        from reflex.event import EventChain
+
+        if event_chain.type_ is not EventChain:
+            raise ValueError(f"Invalid event chain: {event_chain}")
+        return "".join(
+            [
+                "(() => {",
+                format_var(event_chain),
+                f"; preventDefault({format_var(event_arg)})" if event_arg else "",
```

Comment:
The `Event` function in `state.js` should also handle the `preventDefault` logic I thought?

Reply:
if we're formatting an arbitrary var as an event chain, there's no guarantee that `Event` gets called at all; it could be anything.